internal/mongo/client.go

```diff
@@ -88,29 +87,29 @@ func (c *client) Watch(collection *mongodriver.Collection, itemsChan chan *Watch
 
 func (c *client) watchCursor(cursor MongoDriverCursor, itemsChan chan *WatchItem) {
 	for cursor.Next(c.ctx) {
-		var result bson.M
-		if err := cursor.Decode(&result); err != nil {
-			c.logger.Error("Mongo client: Unable to decode bson value from cursor", logger.Error("error", err))
+		var event bson.M
+		if err := cursor.Decode(&event); err != nil {
+			c.logger.Error("Mongo client: Unable to decode change event value from cursor", logger.Error("error", err))
 		}
 
-		if err := c.sendIntoChannel(result, itemsChan); err != nil {
+		if err := c.sendIntoChannel(event, itemsChan); err != nil {
 			c.logger.Error("Mongo client: Unable to send document", logger.Error("error", err))
 		}
 	}
 }
 
-func (c *client) sendIntoChannel(result bson.M, itemsChan chan *WatchItem) error {
-	jsonBytes, err := json.Marshal(result)
+func (c *client) sendIntoChannel(event bson.M, itemsChan chan *WatchItem) error {
+	jsonBytes, err := marshall(event)
 	if err != nil {
-		c.logger.Error("Mongo client: Unable to unmarshal bson to json", logger.Error("error", err))
+		c.logger.Error("Mongo client: Unable to unmarshal change event to json", logger.Error("error", err))
 		return err
 	}
 
-	var oplog OperationLog
-	json.Unmarshal(jsonBytes, &oplog)
+	// if marshalling is ok, the changeEvent is well formed, we can skip error checking
```

Comment:
I think this error should be managed because the first thing that `marshall(event)` function below do is to call a `isValid()` function that calls the `documentID(event)` function and check if it exists or not.

Maybe this code could be moved before calling `marshall(event)` below so you could ensure you have a documentID before marshalling it. If we don't have one, we should not go further

Reply:
Finally I think the best way is to decode the cursor directly into a changeEvent struct.